crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java

```diff
@@ -543,13 +522,7 @@ public Property process() {
           ignored = true;
         }
 
-        if (p.isPreserveUnknownFields()) {
-          preserveUnknownFields = true;
-        }
-
-        if (p.isPreserveSelfUnknownFields()) {
-          preserveSelfUnknownFields = true;
-        }
+        preserveUnknownFields = p.isPreserveUnknownFields();
```

Comment:
If I remember correctly this method can be called multiple times on `object`s, so that the last property that affect the object may reset the flag, the gut feeling is to leave the `if`:

```java
if (p.isPreserveUnknownFields()) {
  preserveUnknownFields = p.isPreserveUnknownFields();
}
```

but haven't checked the code carefully.

Reply:
I just removed the if-else clause because it seemed redundant, but what you say is true. I'll revert this change

Reply:
Looking at it again, and the same comment might apply to the other properties as well (min, max, pattern). 👍

Reply:
You're right. I changed them, please check again :pray:

Reply:
This encoding works for me 👍 
I'm negatively surprised that the test suite is not able to catch a regression here 🙁

Reply:
A next step would be to implement tests that verify the described expected behaviors (multipass processing, etc.)